src/plugins/data_views/common/data_views/_pattern_cache.ts

```diff
@@ -10,30 +10,50 @@ import { DataView } from './data_view';
 
 export interface DataViewCache {
   get: (id: string) => Promise<DataView> | undefined;
+  getByHash: (id: string) => DataView | undefined;
   set: (id: string, value: Promise<DataView>) => Promise<DataView>;
   clear: (id: string) => void;
   clearAll: () => void;
 }
 
 export function createDataViewCache(): DataViewCache {
-  const vals: Record<string, Promise<DataView>> = {};
+  const vals: Map<string, Promise<DataView>> = new Map();
+  const valsByHash: Map<string, DataView> = new Map();
```

Comment:
Main changes are in this file, a bit of refactoring, when introducing a new hash, to use Map(), which wasn't around when this cache was created (I think). However it could also be `{}` like before

Reply:
The use of a Map is a nice improvement, it would probably make sense to do a separate PR just to make sure its not complicating anything.

Reply:
yes for sure!